### ISSUE TITLE
小白求解 swift infer 常规推理模型出现的问题

### ISSUE THREAD

**Issue by ffhelly** [modelscope/ms-swift#3321]
新手咨询。
我按照说明已经安装好整个环境，用的是全功能的安装。
我在测试推理的时候  使用以下的指令

`CUDA_VISIBLE_DEVICES=4 swift infer --model /kiwi/helly/unsloth/model/unsloth/DeepSeek-R1-Distill-Llama-8B/ --max_batch_size 64 --infer_backend vllm --max_length 2048 --stream true --temperature 0.7 --top_k 1 --max_new_tokens 300`

测试的模型是 ds r1的 8b。  

输出为：

`run sh: `/root/miniconda3/envs/msswt/bin/python /kiwi/helly/msswt/ms-swift/swift/cli/infer.py --model /kiwi/helly/unsloth/model/unsloth/DeepSeek-R1-Distill-Llama-8B/ --max_batch_size 64 --infer_backend vllm --max_length 2048 --stream true --temperature 0.7 --top_k 1 --max_new_tokens 300`
[INFO:swift] Successfully registered `/kiwi/helly/msswt/ms-swift/swift/llm/data/dataset_info.json`
[INFO:swift] Start time of running main: 2025-02-28 16:11:22.733780
usage: infer.py [-h] [--model_type MODEL_TYPE] [--model_id_or_path MODEL_ID_OR_PATH] [--model_revision MODEL_REVISION]
                [--sft_type {lora,full,longlora,adalora,ia3,llamapro,vera,boft}] [--template_type TEMPLATE_TYPE]
                [--infer_backend {AUTO,vllm,pt,lmdeploy}] [--ckpt_dir CKPT_DIR] [--result_dir RESULT_DIR]
                [--load_args_from_ckpt_dir [LOAD_ARGS_FROM_CKPT_DIR]] [--no_load_args_from_ckpt_dir]
                [--load_dataset_config [LOAD_DATASET_CONFIG]] [--eval_human EVAL_HUMAN] [--seed SEED] [--dtype {bf16,fp16,fp32,AUTO}]
                [--model_kwargs MODEL_KWARGS] [--dataset DATASET [DATASET ...]] [--val_dataset VAL_DATASET [VAL_DATASET ...]]
                [--dataset_seed DATASET_SEED] [--dataset_test_ratio DATASET_TEST_RATIO] [--show_dataset_sample SHOW_DATASET_SAMPLE]
                [--save_result [SAVE_RESULT]] [--no_save_result] [--system SYSTEM] [--tools_prompt {react_en,react_zh,toolbench}]
                [--max_length MAX_LENGTH] [--truncation_strategy {delete,truncation_left}]
                [--check_dataset_strategy {none,discard,error,warning}] [--model_name MODEL_NAME [MODEL_NAME ...]]
                [--model_author MODEL_AUTHOR [MODEL_AUTHOR ...]] [--quant_method {bnb,hqq,eetq,awq,gptq,aqlm}]
                [--quantization_bit {0,1,2,3,4,8}] [--hqq_axis {0,1}] [--hqq_dynamic_config_path HQQ_DYNAMIC_CONFIG_PATH]
                [--bnb_4bit_comp_dtype {fp16,bf16,fp32,AUTO}] [--bnb_4bit_quant_type {fp4,nf4}]
                [--bnb_4bit_use_double_quant [BNB_4BIT_USE_DOUBLE_QUANT]] [--no_bnb_4bit_use_double_quant]
                [--bnb_4bit_quant_storage BNB_4BIT_QUANT_STORAGE] [--max_new_tokens MAX_NEW_TOKENS] [--do_sample DO_SAMPLE]
                [--temperature TEMPERATURE] [--top_k TOP_K] [--top_p TOP_P] [--repetition_penalty REPETITION_PENALTY]
                [--num_beams NUM_BEAMS] [--stop_words STOP_WORDS [STOP_WORDS ...]] [--rope_scaling {linear,dynamic}]
                [--use_flash_attn USE_FLASH_ATTN] [--ignore_args_error [IGNORE_ARGS_ERROR]] [--stream [STREAM]] [--no_stream]
                [--merge_lora [MERGE_LORA]] [--merge_device_map MERGE_DEVICE_MAP] [--save_safetensors [SAVE_SAFETENSORS]]
                [--no_save_safetensors] [--overwrite_generation_config [OVERWRITE_GENERATION_CONFIG]] [--verbose VERBOSE]
                [--local_repo_path LOCAL_REPO_PATH] [--custom_register_path CUSTOM_REGISTER_PATH]
                [--custom_dataset_info CUSTOM_DATASET_INFO] [--device_map_config DEVICE_MAP_CONFIG]
                [--device_max_memory DEVICE_MAX_MEMORY [DEVICE_MAX_MEMORY ...]] [--hub_token HUB_TOKEN]
                [--gpu_memory_utilization GPU_MEMORY_UTILIZATION] [--tensor_parallel_size TENSOR_PARALLEL_SIZE]
                [--max_num_seqs MAX_NUM_SEQS] [--max_model_len MAX_MODEL_LEN] [--disable_custom_all_reduce [DISABLE_CUSTOM_ALL_REDUCE]]
                [--no_disable_custom_all_reduce] [--enforce_eager [ENFORCE_EAGER]] [--limit_mm_per_prompt LIMIT_MM_PER_PROMPT]
                [--vllm_enable_lora [VLLM_ENABLE_LORA]] [--vllm_max_lora_rank VLLM_MAX_LORA_RANK]
                [--lora_modules LORA_MODULES [LORA_MODULES ...]] [--max_logprobs MAX_LOGPROBS] [--tp TP]
                [--cache_max_entry_count CACHE_MAX_ENTRY_COUNT] [--quant_policy QUANT_POLICY] [--vision_batch_size VISION_BATCH_SIZE]
                [--self_cognition_sample SELF_COGNITION_SAMPLE] [--train_dataset_sample TRAIN_DATASET_SAMPLE]
                [--val_dataset_sample VAL_DATASET_SAMPLE] [--safe_serialization SAFE_SERIALIZATION] [--model_cache_dir MODEL_CACHE_DIR]
                [--merge_lora_and_save MERGE_LORA_AND_SAVE]
                [--custom_train_dataset_path CUSTOM_TRAIN_DATASET_PATH [CUSTOM_TRAIN_DATASET_PATH ...]]
                [--custom_val_dataset_path CUSTOM_VAL_DATASET_PATH [CUSTOM_VAL_DATASET_PATH ...]]
                [--vllm_lora_modules VLLM_LORA_MODULES [VLLM_LORA_MODULES ...]] [--device_map_config_path DEVICE_MAP_CONFIG_PATH]
infer.py: error: ambiguous option: --model could match --model_type, --model-type, --model_id_or_path, --model-id-or-path, --model_revision, --model-revision, --model_kwargs, --model-kwargs, --model_name, --model-name, --model_author, --model-author, --model_cache_dir, --model-cache-dir
`

失败。  是那个参数有问题的吗？   在官方页面上 使用
CUDA_VISIBLE_DEVICES=0 swift infer \
    --model /kiwi/helly/unsloth/model/unsloth/DeepSeek-R1-Distill-Llama-8B  \
    --stream true \
    --infer_backend pt \
    --max_new_tokens 2048

这种方式的推理，也是同样的问题。

**Comment by Jintao-Huang** [comment]
swift版本是什么 升级一下试试

**Comment by ffhelly** [comment]
> swift版本是什么 升级一下试试

ms-swift 2.5  现在好像是这个版本。  直接升级到3.0的吗？

**Comment by ffhelly** [comment]
> swift版本是什么 升级一下试试

git clone https://github.com/modelscope/ms-swift.git
cd ms-swift
pip install -e .

# 全能力
# pip install -e '.[all]'   我是使用这种方式安装的。

**Comment by Jintao-Huang** [comment]
升级到swift3.x.

检查一下pip源的更新是否及时

**Comment by ffhelly** [comment]
> 升级到swift3.x.
> 
> 检查一下pip源的更新是否及时

我看了下我的pip源 是 aliyun的  。  在里面看到的ms-swift中  存在 ms-swift-3.1.1.tar.gz。  但是我在源码安装的时候  pip install -e '.[all]' 一直download的是2.5的版本。 

是跟我的环境有关导致的吗？ 

我当下用那种方式升级到最新比较合适？ 或者说 我需要更换一下相关的pip源。

**Comment by ffhelly** [comment]
> 升级到swift3.x.
> 
> 检查一下pip源的更新是否及时

感谢大佬，我换了3.11的版本 已经可以进行推理了
